{"uuid": "011fb751-27e9-44a4-be81-86620ef3c6c0", "children": ["d04760c7-e691-4bd6-ac88-6d66ddfbd9dd"], "befores": [{"name": "setup", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.SessionNotCreatedException: Message: Expected browser binary location, but unable to find binary in default location, no 'moz:firefoxOptions.binary' capability provided, and no binary flag set on the command line\n\n", "trace": "  File \"C:\\Users\\fujitsu\\Repos\\Karim_Pytest_POM_Framework\\venv\\Lib\\site-packages\\pluggy\\_callers.py\", line 167, in _multicall\n    teardown.throw(outcome._exception)\n    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\fujitsu\\Repos\\Karim_Pytest_POM_Framework\\venv\\Lib\\site-packages\\_pytest\\setuponly.py\", line 36, in pytest_fixture_setup\n    return (yield)\n            ^^^^^\n  File \"C:\\Users\\fujitsu\\Repos\\Karim_Pytest_POM_Framework\\venv\\Lib\\site-packages\\pluggy\\_callers.py\", line 103, in _multicall\n    res = hook_impl.function(*args)\n  File \"C:\\Users\\fujitsu\\Repos\\Karim_Pytest_POM_Framework\\venv\\Lib\\site-packages\\_pytest\\fixtures.py\", line 1139, in pytest_fixture_setup\n    result = call_fixture_func(fixturefunc, request, kwargs)\n  File \"C:\\Users\\fujitsu\\Repos\\Karim_Pytest_POM_Framework\\venv\\Lib\\site-packages\\_pytest\\fixtures.py\", line 890, in call_fixture_func\n    fixture_result = next(generator)\n  File \"C:\\Users\\fujitsu\\Repos\\Karim_Pytest_POM_Framework\\tests\\conftest.py\", line 23, in setup\n    driver = webdriver.Firefox(service=service)\n  File \"C:\\Users\\fujitsu\\Repos\\Karim_Pytest_POM_Framework\\venv\\Lib\\site-packages\\selenium\\webdriver\\firefox\\webdriver.py\", line 71, in __init__\n    super().__init__(command_executor=executor, options=options)\n    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\fujitsu\\Repos\\Karim_Pytest_POM_Framework\\venv\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py\", line 212, in __init__\n    self.start_session(capabilities)\n    ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^\n  File \"C:\\Users\\fujitsu\\Repos\\Karim_Pytest_POM_Framework\\venv\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py\", line 299, in start_session\n    response = self.execute(Command.NEW_SESSION, caps)[\"value\"]\n               ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\fujitsu\\Repos\\Karim_Pytest_POM_Framework\\venv\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py\", line 354, in execute\n    self.error_handler.check_response(response)\n    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^\n  File \"C:\\Users\\fujitsu\\Repos\\Karim_Pytest_POM_Framework\\venv\\Lib\\site-packages\\selenium\\webdriver\\remote\\errorhandler.py\", line 229, in check_response\n    raise exception_class(message, screen, stacktrace)\n"}, "start": 1731177574667, "stop": 1731177585030}], "start": 1731177574667, "stop": 1731177585195}